### ISSUE TITLE
0.2.0 Firmware update has bricked my Arduino R4 Wifi

### ISSUE THREAD

**Issue by princelundgren** [arduino/uno-r4-wifi-usb-bridge#9]
Bug description:
After flashing successfully to 0.2.0. I am no longer able to upload any code to the Arduino R4 Wifi. I am getting error message "No device found on cu.usbmodemDC5475C5165C2". I have tried to flash firmware and upload code from both my Macbook and my Windows PC but I am having the same issue on both.
See below image to see error message I am receiveing when upload code from Arduino IDE.
<img width="926" alt="CleanShot 2023-07-13 at 18 39 46@2x" src="https://github.com/arduino/uno-r4-wifi-usb-bridge/assets/4346800/22416895-3a1a-44a8-a8e0-4ae3939edb57">

Reproduce:
1. Firmware updated using instructions from [this forum post](https://forum.arduino.cc/t/radio-module-firmware-version-0-2-0-is-now-available/1147361). The firmware update is successfully completed. **EDIT: Would like to add, the last step where it says to physically reconnecting the board I am doing as well**
3. Try to Upload code from Arduino IDE but fails with error message "No device found on cu.usbmodemDC5475C5165C2".

**Comment by facchinm** [comment]
Hi @PrinceBalabis ,
Did you disconnect physically the board from the computer after updating the firmware? That's unfortunate but there's a bug in esp32 preventing the board to reboot without user intervention.

**Comment by vshymanskyy** [comment]
Yes, looks like this happens from time to time, not directly after the firmware upgrade.
Reconnecting the board physically helps.

**Comment by PrinceBalabis** [comment]
> 

I have physically reconnected the board to the Macbook after flashing firmware, even tried to connect to my PC afterwards, getting same error message on both computers. To clarify, I can find the board on the list of devices in Arduino IDE on both my Macbook and PC, but it's only when uploading code that I get the error message.

**Comment by LYB030** [comment]
i got same problem

**Comment by PrinceBalabis** [comment]
I bought a 2nd Arduino Uno R4 and tried to flash using the guide on [Arduino Cloud IoT](https://create.arduino.cc/iot/). This broke my 2nd Arduino Uno R4 as well so I have two Arduino Uno R4 which are no longer working.

**Comment by RickorDD** [comment]
i flashed with the Jumper on ESP Pin and flashed with MacOS and ./bin/espflash ... and it was successful
With Windows has not running with error and the workaround with Mac Console was not running without jumper.
After the flash i press the reset button, not USB unplugged

But now I also have the problem that the UNO R4 sometimes does not start immediately and hangs. But after reset button it works

**Comment by pennam** [comment]
@PrinceBalabis @RickorDD are your boards from the early access program?

@PrinceBalabis have you tried using a different PC and/or USB cable? What macOS version are you using?

**Comment by RickorDD** [comment]
no board from early access

**Comment by pennam** [comment]
> After the flash i press the reset button, not USB unplugged

After flashing the wifi firmware you have to unplug the board, remove the jumper wire, and connect the board again to the PC. Only pressing the reset button the board will stay in ESP download mode.

**Comment by PrinceBalabis** [comment]
> @PrinceBalabis @RickorDD are your boards from the early access program?
No


> @PrinceBalabis have you tried using a different PC and/or USB cable? What macOS version are you using?
Yes, I have tried flashing firmware and uploading code from all of these devices:
1.  Macbook #1
2.  Macbook #2
3.  Windows PC

I follow these instructions (including physically disconnecting the USB cable and trying to upload code from all my Macbooks and PC's)
![Screenshot 2023-07-21 203656](https://github.com/arduino/uno-r4-wifi-usb-bridge/assets/4346800/c4c73f7b-0c51-45aa-a150-78b165440242)

See here that the flash is successful from my PC:
![Screenshot 2023-07-21 203623](https://github.com/arduino/uno-r4-wifi-usb-bridge/assets/4346800/d43ca454-c61f-4376-b5b6-a412da3ffbe7)

See here how the upload fails (same with my PC, Mac #1 and Mac #2):
![Screenshot 2023-07-21 203748](https://github.com/arduino/uno-r4-wifi-usb-bridge/assets/4346800/0223d306-8f55-4b2a-b769-c64991b37ece)

**Comment by LYB030** [comment]
same problem

**Comment by RickorDD** [comment]
do you unplugged the jumper? With Jumper is COM5 without COM4

**Comment by RickorDD** [comment]
Update to hang.
UNO R4 starts ever, but sometimes is to long that connect to WiFi
I will check this again.

**Comment by PrinceBalabis** [comment]
> do you unplugged the jumper? With Jumper is COM5 without COM4

I don't use a jumper when flashing 0.2.0. It flashes fine. When I replug the Arduino it is COM4, and when I try to flash it fails. see my previous comment with pictures showing that it is COM5 when flashing and COM4 when uploading code: https://github.com/arduino/uno-r4-wifi-usb-bridge/issues/9#issuecomment-1646102559

**Comment by LYB030** [comment]
<img width="532" alt="image" src="https://github.com/arduino/uno-r4-wifi-usb-bridge/assets/42745293/76438f38-c82f-4e29-a6f0-c794f9df32fd">
same problem!

**Comment by RickorDD** [comment]
when could be use a jumper or when not??? Whats say the Arduino Team? Can you check with jumper?

![Bildschirmfoto 2023-07-22 um 20 01 48](https://github.com/arduino/uno-r4-wifi-usb-bridge/assets/6922544/f9ee7ed2-a8ab-4367-b3a5-199f301e3020)

**Comment by RickorDD** [comment]
![Bildschirmfoto 2023-07-22 um 20 04 51](https://github.com/arduino/uno-r4-wifi-usb-bridge/assets/6922544/da9d0535-a610-443f-b322-9c88e645b9c8)

**Comment by pennam** [comment]
@PrinceBalabis i've prepaerd a package of the `0.1.0` firmware version you can upload using the update script:

* https://github.com/arduino/uno-r4-wifi-usb-bridge/releases/download/0.1.0/unor4wifi-update-macos-0.1.0.zip
* https://github.com/arduino/uno-r4-wifi-usb-bridge/releases/download/0.1.0/unor4wifi-update-linux-0.1.0.zip
* https://github.com/arduino/uno-r4-wifi-usb-bridge/releases/download/0.1.0/unor4wifi-update-windows-0.1.0.zip

can you please check if going back to firmware version 0.1.0 restores upload functionality of your boards?

**Comment by pennam** [comment]
@RickorDD The jumper wire is used to put the board in ESP downaload mode. The same job can be done using the `unor4wifi-reboot-*` binary that is included in the `unor4wifi-update-*` package.

You should not use the jumper wire if you are using the `unor4wifi-update-*` package.

If there are troubles using the `unor4wifi-update-*` package (board is not going in ESP download mode) You can manually put the board in in ESP download mode using the jumper wire following this istructions:

https://github.com/arduino/uno-r4-wifi-usb-bridge/tree/main/unor4wifi-updater#option-2

**Comment by PrinceBalabis** [comment]
> @PrinceBalabis i've prepaerd a package of the `0.1.0` firmware version you can upload using the update script:
> 
> * https://github.com/arduino/uno-r4-wifi-usb-bridge/releases/download/0.1.0/unor4wifi-update-macos-0.1.0.zip
> * https://github.com/arduino/uno-r4-wifi-usb-bridge/releases/download/0.1.0/unor4wifi-update-linux-0.1.0.zip
> * https://github.com/arduino/uno-r4-wifi-usb-bridge/releases/download/0.1.0/unor4wifi-update-windows-0.1.0.zip
> 
> can you please check if going back to firmware version 0.1.0 restores upload functionality of your boards?

Thank you for the help! Can confirm I flashed both my Arduinos R4's and I can now upload code just fine.

1. Downloaded the macos 0.1.0 zip file you linked
2. Ran chmod +x update.command
3. Ran ./update.command. 
4. The flash was successful
5. Unplug and replug Arduino R4
6. Upload code from Arduino IDE
7. Upload successful!

<img width="571" alt="CleanShot 2023-07-25 at 01 53 36@2x" src="https://github.com/arduino/uno-r4-wifi-usb-bridge/assets/4346800/23beb23e-59d5-4e3e-9600-f1969fa93bb2">

<img width="805" alt="CleanShot 2023-07-25 at 01 54 32@2x" src="https://github.com/arduino/uno-r4-wifi-usb-bridge/assets/4346800/f0fa671a-c604-4d18-a633-50795617ea28">

**Comment by PrinceBalabis** [comment]
Now this is interesting. After flashing to 0.1.0, I thought I'll try to flash 0.2.0 again. I did the same procedure and now uploading code to my 0.2.0 Arduino works fine. I tried the same with flashing my 2nd Arduino, and it works fine too.

I can only think of two theories why my Arduinos bricked:
1. I had almost the exact same code on both my Arduino R4's when I flashed them (both Arduinos completely unattached from any components other than the USB cable so it wasn't hardware related). Perhaps the uploaded code somehow affected the flash and caused the brick. However, this doesn't exactly explain why the Arduinos started working again after flashing 0.1.0, after trying to reflash 0.2.0 multiple times and they remained bricked. So this theory is maybe less likely.
2. Flashing 0.2.0 on a vanilla production Arduino R4 bricks it. Flashing with the 0.1.0 firmware first somehow allows for 0.2.0 to flash successfully.

**Comment by pennam** [comment]
@PrinceBalabis Thanks for the feedback `0.1.0` is the vanilla production firmware so i suspect the issue could be in the sketch loaded during the upgrade from `0.1.0` to `0.2.0`. Do you remember what sketch was running when your board has bricked?

**Comment by PrinceBalabis** [comment]
> @PrinceBalabis Thanks for the feedback `0.1.0` is the vanilla production firmware so i suspect the issue could be in the sketch loaded during the upgrade from `0.1.0` to `0.2.0`. Do you remember what sketch was running when your board has bricked?

I'm not completely sure of the exact code because I don't have a commits on the code before flashing, just approximately.
I had my own code uploaded used for a quadcopter project, though because it was very early stage of development, it was not a lot of code involved when I was flashing the first Arduino, and looking back at the commit history, I believe it was quite a different code from when I flashed the 2nd Arduino.

I believe when I flashed the first Arduino, it was quite lightweight, it involved Serial, FreeRTOS & blinking built-in LED, I might've had some WiFi code then, not sure. 

The code I had uploaded when I flashed the 2nd Arduino, No longer had FreeRTOS but included Wire.h and some code for communicating with MPU-9150 and MP-280.
